### PR TITLE
replace remaining blocking code, #75

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #75

File: src/main/scala/akka/persistence/cassandra/CassandraPluginConfig.scala [hunk 1]
```diff
@@ -53,10 +53,6 @@ class CassandraPluginConfig(system: ActorSystem, config: Config) {
   val writeRetries: Int = config.getInt("write-retries")
   val readRetries: Int = config.getInt("read-retries")
 
-  val blockingDispatcherId: String = config.getString("blocking-dispatcher")
-
-  // FIXME temporary until we have fixed all blocking
-  private[cassandra] val blockingTimeout: FiniteDuration = 10.seconds
```
Comment (ktoso) [inline src/main/scala/akka/persistence/cassandra/CassandraPluginConfig.scala]: 👍

File: src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala [hunk 2]
```diff
@@ -86,8 +90,9 @@ class CassandraSnapshotStore(cfg: Config) extends SnapshotStore with CassandraSt
 
   override def loadAsync(persistenceId: String, criteria: SnapshotSelectionCriteria): Future[Option[SelectedSnapshot]] = {
     for {
-      prepStmt <- preparedSelectSnapshotMetadataForLoad
-      mds <- metadata(prepStmt, persistenceId, criteria).map(_.take(maxLoadAttempts))
+      prepStmt <- if (criteria.maxTimestamp == Long.MaxValue) preparedSelectSnapshotMetadataWithMaxLoadAttemptsLimit
+      else preparedSelectSnapshotMetadata
```
Comment (johanandren) [inline src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala]: I'm not sure I get this `Long.MaxValue` is used as a magic value to select a different select here?

Also: I'd like this better with a linefeed and both branches in the same intendentation level.

File: src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala [hunk 3]
```diff
@@ -162,8 +167,8 @@ class CassandraSnapshotStore(cfg: Config) extends SnapshotStore with CassandraSt
   }
 
   override def deleteAsync(persistenceId: String, criteria: SnapshotSelectionCriteria): Future[Unit] = {
-    preparedSelectSnapshotMetadataForDelete.flatMap { prepStmt =>
-      metadata(prepStmt, persistenceId, criteria).flatMap { mds =>
+    preparedSelectSnapshotMetadata.flatMap { prepStmt =>
+      metadata(prepStmt, persistenceId, criteria, limit = -1).flatMap { mds =>
```
Comment (johanandren) [inline src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala]: Yuck, another magic number! Can we do it with `Option` instead, or is this to save allocs?

File: src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala (hunk 2 above)
Comment (patriknw) [inline src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala]: fixed, and added comments

File: src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala (hunk 3 above)
Comment (patriknw) [inline src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala]: good point, no need for magic here, Option it is

**Pull request merged**